scripts/devops_tasks/python_snippet_updater.py

```diff
@@ -0,0 +1,94 @@
+import sys
+from pathlib import Path
+import re
+
+snippets = {}
+not_up_to_date = False
+
+target_snippet_sources = ["samples/*.py", "samples/*/*.py"]
+target_md_files = ["README.md"]
+
+def get_snippet(file):
+    with open(file, 'r') as f:
+        content = f.read()
+    pattern = "# \\[START[A-Z a-z0-9_]+\\][\\s\\S]+?# \\[END[A-Z a-z0-9_]+\\]"
+    matches = re.findall(pattern, content)
+    for match in matches:
+        s = match
+        pos1 = s.index("[START")
+        pos2 = s.index("]")
+        name = s[pos1 + 6:pos2].strip()
+        s = s[pos2 + 1:]
+        pos1 = s.index("# [END")
+        snippet = s[:pos1-1]
+        # Remove extra spaces
+        spaces = ""
+        for char in snippet[1:]:
+            if char == " ":
+                spaces += char
+            else:
+                break
+        snippet = snippet.replace("\n" + spaces, "\n")
+        # Remove first newline
+        snippet = snippet[1:].rstrip()
+        if snippet[-1] == "\n":
+            snippet = snippet[:-1]
+
+        file_name = str(file.name)[:-3]
+        identifier = ".".join([file_name, name])
+        if identifier in snippets.keys():
+            print(f'Warning: found duplicated snippet name "{identifier}".')
+            print(file)
+        # print(f"Found: {file.name}.{name}")
+        snippets[identifier] = snippet
+
+
+def update_snippet(file):
```

Comment:
Any new scripts or updates, please add `type annotations`.

Reply:
After the big refactor that dropped py2.7 I don't add EngSys code that isn't at least type annotated and doc-commented. 👍